### ISSUE TITLE
[BUG] Button labels do not update after invoking callbacks with bokeh >= 2.*.*

### ISSUE THREAD

**Issue by AlexHiesch** [bokeh/bokeh#10134]
OS and Software RHEL, python 3.6, firefox 77.0.1, Chrome 83.0.4103.97, IE 11, bokeh server

Callback events do not update button labels with bokeh >= 2.*.*
Code runs perfectly with bokeh 1.4.0

#### Complete, minimal, self-contained example code that reproduces the issue

```
from bokeh.io import curdoc
from bokeh.io import curdoc, show
from bokeh.layouts import row, widgetbox, gridplot
from bokeh.models import ColumnDataSource, DataTable
from bokeh.models.widgets import Slider, TextInput, Div, Button, Slider
from bokeh.models.widgets import CheckboxButtonGroup, DatePicker, RadioButtonGroup
from bokeh.models.widgets import CheckboxGroup, RadioGroup
from bokeh.models.widgets import TableColumn, Select, Dropdown
from bokeh.layouts import gridplot, grid, widgetbox, column
from bokeh.events import ButtonClick
import time

def predict(event):
    i=0
    for i in range(10):
        print("Clicked Predict Button")
        button_predict.button_type = "warning"
        button_predict.label = "Computing started for loop {}".format(i)
        time.sleep(10)

div_text_intro = Div(text= """Reproduce Bug with non-updatable Button labels""")

# Predict Button
button_predict = Button(label="Predict", button_type="success")
# Register callback
button_predict.on_event(ButtonClick, predict)

curdoc().add_root(div_text_intro)
curdoc().add_root(button_predict)

```

#### Stack traceback and/or browser JavaScript console output
unfortunately there are no errors in python or browser dev consoles (css, js)

**Comment by bryevdv** [comment]
@AlexHiesch Python <--> JS synchronization only happens at function boundaries, so your updates are blocked by the `sleep` callsyou make. Remove those and you will see the button update immediately. To do this you will need to use the `add_next_tick_callback` approach that is described in a few places both on SO and the project discourse. Also happy to continue discussing usage details directly on the Discourse:

https://discourse.bokeh.org/

Now that Bokeh is Python 3.6+ and moved to native co-reoutines, we hope to make it simpler to do things like this with async callbacks that synchronize Python <--> JS on `await` boundaries as well as function boundaries. But that is future work.
